iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/memtable/TsFileProcessor.java

```diff
@@ -577,6 +597,16 @@ public void insertTablet(
     if (!insertTabletNode.isGeneratedByPipe()) {
       workMemTable.markAsNotGeneratedByPipe();
     }
+    final String originClusterId =
+        insertTabletNode.getOriginClusterId() == null
+            ? config.getClusterId()
+            : insertTabletNode.getOriginClusterId();
+    if (Objects.isNull(workMemTable.getCurrentOriginClusterId())) {
+      workMemTable.setCurrentOriginClusterId(originClusterId);
+    } else if (!Objects.equals(originClusterId, workMemTable.getCurrentOriginClusterId())) {
```

Comment:
originClusterId != workMemTable.getCurrentOriginClusterId()

Reply:
Make a static final cache map in the receiver and get the id string from the map

map <id, id> putIfAbsent

Reply:
comment here if using the ref comparing